examples/generate_bom/compute_cogs.py

```diff
@@ -0,0 +1,205 @@
+
+    :param part_number: A part number by which to search for the component.
+    :returns: A mapping of price breakpoints to the price at that breakpoint.
+    """
+
+    if part_number.startswith("NOTAPART"):
+        return {}
+
+    api_key = os.environ.get("COFACTR_API_KEY")
+    client_id = os.environ.get("COFACTR_CLIENT_ID")
+    if api_key is None or client_id is None:
+        raise ValueError(
+            "Please set the COFACTR_API_KEY and COFACTR_CLIENT_ID environment variables"
+        )
+
+    seach_response = requests.get(
+        "https://graph.cofactr.com/products/",
+        headers={
+            "X-API-KEY": api_key,
+            "X-CLIENT-ID": client_id,
+        },
+        params={
+            "q": part_number,
+            "schema": "product-offers-v0",
+            "external": True,
+            "limit": 1,
+        },
+    )
+
+    if seach_response.status_code != 200:
+        print(
+            f"Warning: Received status code {seach_response.status_code} for {part_number}",
+            file=sys.stderr,
+        )
+        return {}
+
+    search_results = seach_response.json()
+    try:
+        reference_prices = search_results.get("data", [])[0].get("reference_prices")
+    except IndexError:
+        print(
+            f"Warning: No results found for {part_number}",
+            file=sys.stderr,
+        )
+        return {}
+
+    prices = {
+        int(price["quantity"]): float(price["price"]) for price in reference_prices
+    }
+
+    return prices
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "bom_file",
+        help="The path to the BOM file.",
+    )
+    parser.add_argument(
+        "--quantities",
+        help=(
+            "A comma-separated list of quantities of PCBs to compute the COGS "
+            + "for. E.g. 1,10,100,1000. Defaults to the example."
+        ),
+        default="1,10,100,1000",
+    )
+    parser.add_argument(
+        "--output-file",
+        help="The path to the output file. Defaults to stdout, i.e. printing to the console.",
+    )
+
+    args = parser.parse_args()
+
+    quantities = [int(quantity) for quantity in args.quantities.split(",")]
+
+    with open(args.bom_file, "r") as bom_file:
+        bom_csv = csv.DictReader(bom_file)
+
+        parts = [
+            part
+            for part in bom_csv
+            if part[PART_NUMBER_COLUMN] and part[PART_NUMBER_COLUMN] != ""
+        ]
+
+    print(f"Computing COGS for {len(parts)} parts", file=sys.stderr)
+    print(f"Fetching prices for {len(parts)} parts", file=sys.stderr)
+
+    prices_for_parts = {}
+
+    for part in parts:
+        part_number = part[PART_NUMBER_COLUMN]
+        prices = fetch_price_for_part(part_number)
+        if prices and len(prices) > 0:
+            prices_for_parts[part_number] = fetch_price_for_part(part_number)
+
+    print(f"Found prices for {len(prices_for_parts)} parts", file=sys.stderr)
```

Comment:
Grammar when length = 1. When I don't want to bother with pluralization, I just flip it around to say: `f'Number of prices found: {n}'`